src/back-end/lib/resources/affiliation.ts

```diff
@@ -131,40 +124,38 @@ const resource: Resource = {
 
   update(connection) {
     return {
-      async parseRequestBody(request) {
+      async parseRequestBody(request): Promise<null> {
         return null;
       },
-      async validateRequestBody(request) {
-        const validatedAffiliationId = await validateAffiliationId(connection, request.params.id);
-        if (validatedAffiliationId.tag === 'invalid') {
+      async validateRequestBody(request): Promise<Validation<ValidatedUpdateRequestBody, UpdateValidationErrors>> {
+        const validatedAffiliation = await validateAffiliationId(connection, request.params.id);
+        if (isInvalid(validatedAffiliation)) {
           return invalid({
-            affiliation: getInvalidValue(validatedAffiliationId, undefined)
+            affiliation: getInvalidValue(validatedAffiliation, undefined)
           });
         }
-        const existingAffiliation = validatedAffiliationId.value;
+        const existingAffiliation = validatedAffiliation.value;
         if (existingAffiliation.membershipStatus === MembershipStatus.Pending) {
           if (!permissions.updateAffiliation(connection, request.session, existingAffiliation.organization.id)) {
```

Comment:
You need to `await permissions.updateAffiliation...` because it returns a promise.